src/Product/Attributes/AttributeManager.php

```diff
@@ -78,13 +79,16 @@ public function update( WC_Product $product, AttributeInterface $attribute ) {
 	public function get( WC_Product $product, string $attribute_id ): ?AttributeInterface {
 		$this->validate( $product, $attribute_id );
 
-		$attribute_class = $this->get_attribute_types_for_product( $product )[ $attribute_id ];
-		$value           = get_post_meta( $product->get_id(), $this->prefix_meta_key( $attribute_id ), true );
+		$value = null;
+		if ( $product->meta_exists( $this->prefix_meta_key( $attribute_id ) ) ) {
+			$value = $product->get_meta( $this->prefix_meta_key( $attribute_id ), true );
+		}
 
 		if ( null === $value || '' === $value ) {
 			return null;
 		}
 
+		$attribute_class = $this->get_attribute_types_for_product( $product )[ $attribute_id ];
```

Comment:
It's validated in the same method so it shouldn't fail. The `attribute_id` exists in the array or else it will throw an `InvalidValue` exception.

https://github.com/woocommerce/google-listings-and-ads/blob/41f856bc079c91fc29ac2278fa41d8fe20f32f16/src/Product/Attributes/AttributeManager.php#L80